backend/backend-standard-and-best-practices.md

```diff
@@ -0,0 +1,278 @@
+  "last_name": "My last name",
+  "email": "My email"
+}
+```
+
+instead of:
+
+```json
+{
+  "user": {
+    "id": 1,
+    "first_name": "My first name",
+    "last_name": "My last name",
+    "email": "My email"
+  }
+}
+```
+
+## 3- Response bodies
+
+### 3.1- Pagination
+
+All paginated lists requested to our APIs should be returned using the following structure:
+
+```json
+{
+ "page": [
+    {
+      "id": 4,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    },
+    {
+      "id": 5,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    },
+    {
+      "id": 6,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    }
+ ],
+ "count": 3,
+ "limit": 3,
+ "offset": 0,
+ "total_pages": 4,
+ "total_count": 12,
+ "previous_page": 1,
+ "current_page": 2,
+ "next_page": 3,
+ "previous_page_link": "domain.com/users?page=1&limit=3",
+ "current_page_link": "domain/com/users?page=2&limit=3",
+ "next_page_link": "domain.com/users?page=3&limit=3",
+ "first_page_link": "domain.com/users?page=1&limit=3",
+ "last_page_link": "domain.com/users?page=4&limit=3",
+ "n_page_link": "domain.com/users?page=X&limit=3"
+}
+```
+
+Some clarifications about the pagination response:
+- The starting page number should always be 1.
+- When returning the first page, the *previous_page* property should be `NULL`.
+- The parameters with the `_link` suffix are suggested when using link-oriented pagination; otherwise, they should not be included in the response body.
+
+### 3.2- Error forwarding
+
+When informing a client of a request error, the response body should be returned using the following structure:
+
+```json
+{
+  "errors": [
+    { "code": "111", "message": "age must be an int" }
+    { "code": "112", "message": "email is mandatory" }
+  ],
+  "origin": "api-name",
+  "stack_trace": "...",
+  "timestamp": "2019-09-10 03:14:07"
+}
+```
+
+Some clarifications about the error response:
+- The `timestamp` field is not mandatory, but its use is recommended for logging purposes.
+- Both the `origin` and `stack_trace` fields are considered sensitive information, and they should not be sent to a frontend or external client. Their use should be limited between internal APIs of the same application.
+- The `code` field in the errors list is intended as a unique identifier of the returned error message, and a way of keeping an internal record of all errors handled by the application. Also, it should not be a random value, but instead follow a certain convention (for example, codes with format `01XX` could be used for invalid request errors, `02XX` for application errors, etc.). We will define a standard convention for error codes in future versions of this document.
+- Depending on the case and the receiver of the response, it may not be necessary to send both the `code` and `message` parameters.
+
+## 4- API Versioning
```

Comment:
Maybe we can add an observation here saying that the idea is to avoid having a lot of versions alive at the same time, because this is difficult to maintain

Reply:
Great suggestion! Im adding it